### ISSUE TITLE
[Feature request] Include query results data into alert messages

### ISSUE THREAD

**Issue by zax-slaughter** [grafana/grafana#7690]
- Grafana version 4.1.1
- InfluxDB
- Ubuntu 14

- What I did is I tried to set up sensible alerts on many servers. I wanted to have a single query report failures on many servers. Query would return a value based on which notification is sent (say count of errors) + tag information that would allow me to identify the error type and specific server failing.

- What I expected it that it'll be absolutely awesome as all the rest of grafana. However alerts don't work with templates (I know there is another issue on that), so I had to put all errors into a single query. Now the problem is that with many servers there is no way to have meaningful notification sent. Meaningful notification would include information on the errors, such as tags, or error message, ect. That would allow to direct it to a right support group.

- What happens instead, is that I have to set up individual alert panel for each new server, otherwise I can't send meaningful notifications. That of course allows to switch notifications on and off easily with a high fidelity, but requires a lot of panel duplication that is tedious and could have been avoided.

I think this is a relatively small and useful request to allow notification message to include
- alert condition (say we are alerting on count(*) - include threshold end count(*))
- same as alias by allows, such as $tag_system or $tag_error_type, etc

**Comment by daniellee** [comment]
Possible dupe: #5848

**Comment by zax-slaughter** [comment]
I would say that #7121 <https://github.com/grafana/grafana/issues/7121> is
closer, but regardless, I am asking for much smaller feature to make the
feature usable by real world support.

**Comment by nikskiz** [comment]
This feature will be awesome

**Comment by beingyash** [comment]
This is definitely would be very helpful.

**Comment by Zappes** [comment]
This would actually make thed alerts helpful, which they are currently not.

**Comment by ysoftman** [comment]
Isn't this feature completed yet?
(I'm using Grafana version 7.4.5, How can I include query results into the alert notification message?)

**Comment by rlevytskyi** [comment]
Hi All,
Is this feature implemented?
Or maybe there is other way to get some data from query results and pass it to the alert message?

**Comment by ceeeekay** [comment]
Adding to this issue, because I've just spent ages trying to set up a PoC to extract some useful string data for use in alert labels, only to find out the developers don't think doing such a thing is worthwhile.

It's astounding that something so obviously useful has been overlooked (or just straight up denied) by the developers for so long. Alerts _need_ to be enriched with relevant supporting data so that the people receiving them have at least some idea what's going on, instead of dealing with some inscrutable message at 4am.

Our environment has over 2000 hosts. There's absolutely no way that creating _at least_ that many customised alerts is feasible or sustainable. Even just being able to include e.g., `hostname`, `service` and `status` in an alert would mostly allow the creation of several standardised alert rules and drastically reduce the admin overhead required to support more than a handful of servers.

tl;dr: How is this not a feature already?

**Comment by KaplanSarah** [comment]
This functionality is available in Unified Alerting through templating.  Note the 'Values' field in the link below.
https://grafana.com/docs/grafana/latest/alerting/manage-notifications/template-notifications/reference/

**Comment by ChristianHoumann** [comment]
Im having trouble finding out how to make this work through templating. I have a query that finds the string name of some devices with a specific failed state and returns them. The alert query already works through returning 1 if there are found any names, but the problem is that i cannot see which devices are failed, only that one has. I want another query in the alert that returns the names themselves so i can see what devices to look at. So is there any way to extract a string field from a query into a notification template without it failing because there is a string value from the query?

**Comment by DugeraProve** [comment]
@KaplanSarah I'm unclear why you've closed this issue? The Values field doesn't work for a query that can return string results.

I have a similar situation to @ceeeekay where I have one alert that should trigger for any/all instances that meet a condition. However I cannot using Values include the instances that trigger for the condition in the Annotations.

I could create individual alerts for every single instance and if it meets a condition it triggers, but then you go from one alert for the condition to n amount of alerts for the condition where n is the amount of instances you need to monitor.

I'd ask that this issue be re-opened as the issue hasn't actually been address

**Comment by DugeraProve** [comment]
I see from this issue https://github.com/grafana/grafana/issues/72172 my problem was using Metric Search instead of Metric Query with the cloudwatch datasource. Once switching to Metric Query and following the pattern in the linked issue I was able to get things to play nicely and use the Dimension Key Values from Cloudwatch.

**Comment by alext-extracellular** [comment]
For anyone else looking, this appears to work, but it depends on your datasource. some will give the wide vs long error, and some will work. 
eg: https://community.grafana.com/t/mysql-columns-as-labels-for-alerting/91255/6
